include/deal.II/particles/property_pool.h

```diff
@@ -115,14 +124,37 @@ namespace Particles
      */
     const unsigned int n_properties;
 
+    /**
+     * The currently allocated properties (whether assigned to
+     * a particle or available for assignment).
+     */
+    std::vector<double> properties;
+
     /**
      * A collection of handles that have been created by
-     * allocate_properties_array() and have not been destroyed by
-     * deallocate_properties_array().
+     * allocate_properties_array() and have been destroyed by
+     * deallocate_properties_array(). Since the memory is still
+     * allocated these handles can be reused for new particles
+     * to avoid memory allocation.
      */
-    std::set<Handle> currently_open_handles;
+    std::vector<Handle> currently_available_handles;
   };
 
+    /* ---------------------- inline and template functions ------------------ */
+
+  inline ArrayView<double>
+  PropertyPool::get_properties(const Handle handle)
+  {
+    const std::vector<double>::size_type data_index =
+      (handle != invalid_handle) ? handle * n_properties : 0;
```

Comment:
Ah, I see, you're already doing the "index blocks" as opposed to "index elements" thing. Good work!

Reply:
Actually, what you're still missing here is an `Assert(handle != invalid_handle, ...);`. Nothing good can come off presenting an invalid handle here. If you add that assertion, you don't need the `?:` expression.